Fix/handle line parsing, preserves json

harrisonablack opened the pull request:
ELCI supports doCommand, however when a Minecraft command is passed with json arguments, it breaks as the handleLine function split inputing using a comma ",". This PR fixes that behaviour by appropriately parsing the JSON string, preserving commas

harrisonablack commented:
@rozukke

rozukke commented:
From a personal perspective, I would much rather some form of rudimentary KLV encoding to fix the root problem rather than piling things on to what is and was an ad-hoc encoding solution from the previous python impl. 

Not sure how interested you are in trying that out, but I'm also not currently involved much in the decision-making at the C++ Studio course; that solution may have its own set of issues such as leadership not wanting to make risky code changes for edge cases.

dxrcy commented:
Hi @harrisonablack, I have a couple of clarifying questions:
1. Is `doCommand` the only ELCI command (you have used) which is affected by this problem?
2. Are commas the only character which causes this problem?
3. Are the updates to `pom.xml` required for this change?

harrisonablack commented:
Hey @dxrcy,
1. `doCommand` is the only ELCI command which breaks as ELCI never expected arguments to contain commas as part of the command / argument as every other argument / command is alphanumeric
2. Commas are the only character that breaks as the handleLine uses commas as a delimiter, separating arguments 
```java
// RemoteSession.java :116
protected void handleLine(String line) {
		String methodName = line.substring(0, line.indexOf("("));
    // Delimits using .split(",")
		String[] args = line.substring(line.indexOf("(") + 1, line.length() - 1).split(",");
		handleCommand(methodName, args);
	}
```
3. Upon further realisation no. The updates to the Pom.xml aren't required, however i did have to downgrade my JDK to compile the plugin as i was using a JDK that didn't supporting targeting Java 7
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.0.2:compile (default-compile) on project ELCI: Compilation failure: Compilation failure:
[ERROR] error: Source option 7 is no longer supported. Use 8 or later.
[ERROR] error: Target option 7 is no longer supported. Use 8 or later.
```

twhlynch commented:
I too am in this conversation

dxrcy commented:
@harrisonablack 
In that case, I think the better option would be to re-join the arguments when they are used in the `player.doCommand` case, similar to how it's done for `chat.post` (which supports commas completely as far as I can tell). Have you found any issues with this option?

```diff
diff --git a/src/main/java/net/rozukke/elci/RemoteSession.java b/src/main/java/net/rozukke/elci/RemoteSession.java
index 19eab2b..d66cd56 100644
--- a/src/main/java/net/rozukke/elci/RemoteSession.java
+++ b/src/main/java/net/rozukke/elci/RemoteSession.java
@@ -142,7 +142,7 @@ public class RemoteSession {
 					if (player==null) {
 						break;
 					}
-					player.performCommand(args[0]);
+					player.performCommand(String.join(",", args));
 					break;
 				}
 				case "world.getBlock": {
```

harrisonablack commented:
@dxrcy 
Just compiled and tested briefly ... Seems to be working perfectly, fixed the original problem i had. Makes my implementation look way too overly convoluted for what it needed to be ahahah.

Commit:
fix(handleLine-parsing): Re-apend args using delimited comma

@dxrcy Suggested implementation to fix Json Parsing

dxrcy commented:
@harrisonablack thanks :D
